src/ingest-pipeline/airflow/dags/utils.py

```diff
@@ -1238,6 +1238,218 @@ def send_status_msg(**kwargs) -> bool:
+
+    'dataset_uuid_fun' is a function which returns the uuid of the dataset to be
+    updated, or None.  If given, it will be called with **kwargs arguments.
+
+    'dataset_lz_path_fun' is a function which returns the full path of the dataset
+    data directory, or None.  If given, it will be called with **kwargs arguments.
+    If the return value of this callable is None or the empty string, no file metadata
+    will be ultimately be included in the status message.
+
+    'uuid_src_task_id' is the Airflow task_id of a task providing the uuid via
+    the XCOM key 'derived_dataset_uuid' and the dataset data directory
+    via the None key.  This is used only if dataset_uuid is None or dataset_lz_path
+    is None.
+
+    'metadata_fun' is a function which returns additional metadata in JSON form,
+    or None.  If given, it will be called with **kwargs arguments.  This function
+    will only be evaluated if retcode_ops have all returned 0.
+
+    'include_file_metadata is a boolean defaulting to True which indicates whether
+    file metadata should be included in the transmitted metadata structure.  If False,
+    no file metadata will be included.  Note that file metadata may also be excluded
+    based on the return value of 'dataset_lz_path_fun' above.
+    """
+
+    # Does the string represent a "true" value, or an int that is 1
+    def __is_true(val):
+        if val is None: return False
+        if isinstance(val, str):
+            uval = val.upper().strip()
+            if uval in ['TRUE', 'T', '1', 'Y', 'YES']:
+                return True
+            else:
+                return False
+        elif isinstance(val, int) and val == 1:
+            return True
+        else:
+            return False
+
+    def send_status_msg(**kwargs) -> bool:
+        retcodes = [
+            kwargs['ti'].xcom_pull(task_ids=op)
+            for op in retcode_ops
+        ]
+        retcodes = [int(rc or '0') for rc in retcodes]
+        print('retcodes: ', {k: v for k, v in zip(retcode_ops, retcodes)})
+        success = all(rc == 0 for rc in retcodes)
+        if dataset_uuid_fun is None:
+            dataset_uuid = kwargs['ti'].xcom_pull(
+                key='derived_dataset_uuid',
+                task_ids=uuid_src_task_id,
+            )
+        else:
+            dataset_uuid = dataset_uuid_fun(**kwargs)
+        if dataset_lz_path_fun is None:
+            ds_dir = kwargs['ti'].xcom_pull(task_ids=uuid_src_task_id)
+        else:
+            ds_dir = dataset_lz_path_fun(**kwargs)
+        endpoint = '/entities/' + dataset_uuid
+        method = 'PUT'
+        headers = {
+            'authorization': 'Bearer ' + get_auth_tok(**kwargs),
+            'content-type': 'application/json',
+            'X-Hubmap-Application': 'ingest-pipeline',
+        }
+        extra_options = {}
+        return_status = True  # mark false on failure
+
+        http_hook = HttpHook(method, http_conn_id=http_conn_id)
```

Comment:
Are there any places in the DAGs where http_conn_id is explicitly passed in?

Reply:
No, will remove that from the function since it is not used anymore.